sigstore/_internal/merkle.py

```diff
@@ -133,3 +134,33 @@ def verify_merkle_inclusion(entry: LogEntry) -> None:
             f"Inclusion proof contains invalid root hash: expected {inclusion_proof}, calculated "
             f"{calc_hash}"
         )
+
+
+def verify_checkpoint(client: RekorClient, entry: LogEntry) -> None:
+    """
+    Verify the inclusion proof's checkpoint.
+    """
+
+    inclusion_proof = entry.inclusion_proof
+    if inclusion_proof is None:
+        raise InvalidInclusionProofError("Rekor entry has no inclusion proof")
+    if inclusion_proof.checkpoint is None:
+        return
+
+    # verififcaiton occurs in two stages:
+    # 1) verify the signature on the checkpoint
+    # 2) verify the root hash in the checkpoint matches the root hash from the inclusion proof.
+
+    signed_checkpoint = SignedCheckpoint.from_text(inclusion_proof.checkpoint)
+
+    # FIXME(jl): produces an invalid signature exception -- am I using the keyring correctly?
+    # signed_checkpoint.signed_note.verify(client)
```

Comment:
I might be missing it, but where is this `verify` method defined? I don't see it declared on `SignedNote`.